src/lib/ext-activitylog.js

```diff
@@ -4,25 +4,58 @@ class Model {
   constructor() {
     this.logs = [];
     this.filter = {
-      id: [],
-      viewType: [],
-      type: [],
+      id: new Set(),
+      viewType: new Set(),
+      type: new Set(),
+      keyword: '',
     };
   }
 
   addNewLogs(logs) {
     this.logs.push(...logs);
   }
 
-  addFilter({ logKey, valueEquals }) {
-    this.filter[logKey].push(valueEquals);
+  /**
+   * @param {object} updateFilter - It contains key and value to replace
+   * the filter
+   * @param {Set<string>} [updateFilter.id] - It contains the extension ids.
+   * @param {Set<string>} [updateFilter.viewType] - It contains view types that
+   * includes background, popup, sidebar, tab, devtools_page, devtools_panel.
+   * @param {undefined} [updateFilter.viewType] - when [updateFilter.type] is
```

Comment:
Don't create separate definitions for the same property. You can use `Set<string|undefined>` to merge the types. If you feel a need to explain the `undefined` value, do so in the description.

Reply:
done :-)